### PR TITLE
Improve type handling in usage of Jackson

### PR DESCRIPTION
**Pull request by mnonnenmacher**
Please see the commit messages for details.

---
This change is [Reviewable](https://reviewable.io/reviews/heremaps/oss-review-toolkit/928)

### CODE REVIEW

File: model/src/main/kotlin/OutputFormat.kt [hunk 1]
```diff
@@ -52,6 +53,6 @@ fun File.mapper() =
         )
 
 /**
- * Use the mapper returned from [File.mapper] to read the [valueType] object from this file using Jackson.
+ * Use the Jackson mapper returned from [File.mapper] to read an object of type [T] from this file.
  */
-fun <T> File.readValue(valueType: Class<T>): T = mapper().readValue(this, valueType)
+inline fun <reified T : Any> File.readValue(): T = mapper().readValue(this)
```
Comment (sschuberth) [inline model/src/main/kotlin/OutputFormat.kt]: Nit: Space before `:`.

Comment (mnonnenmacher) [reply model/src/main/kotlin/OutputFormat.kt]: That's in line with the Kotlin coding conventions (space before colon if it separates type and supertype):
https://kotlinlang.org/docs/reference/coding-conventions.html#colon

Comment (sschuberth) [reply model/src/main/kotlin/OutputFormat.kt]: Wow, surprising, good to know!

**Comment by sschuberth** [comment]
Merging this despite the unrelated AppVeyor CI failure.

**Pull request merged**